### DISCUSSION TITLE
리더보드 추가와 관련된 문제가 있습니다.

### DISCUSSION THREAD

**Discussion by kihye** [naverz/zepeto-studio-kor#424] [5. Q&A]
![image](https://user-images.githubusercontent.com/80503690/155293466-12085446-94cd-4f5b-a37b-98e96a56071e.png)
리더보드를 사용하려는데 거의 한 시간을 기다려도 그냥 저 상태로 유지만 되고 진전은 없네요..
강제로 껐다 다시 진행도 시켜보고, 프로젝트도 아예 새로 만들어서 해도 결국 추가가 안되는 건 똑같습니다.

도대체 뭐가 문제인지 모르겠습니다.
좋은 해결 방법이 있을까요...

**Comment by x4cf** [comment]
저도 가끔 되고 가끔 오래 로딩되고 그러더라구요

**Reply by kihye** [reply]
그렇다고 마냥 기다리기엔 작업자체가 프리징되니 답이 없더라구요 ㅠㅠ

**Comment by cheolhui-lee** [comment]
안녕하세요.
혹시 사용 중이신 패키지들의 버전 정보를 알 수 있을까요?

**Reply by kihye** [reply]
![image](https://user-images.githubusercontent.com/80503690/155667236-4f79dfce-f451-4dd2-8fce-6f71ef49cbb1.png)
제페토 관련으론 이게 전부인데 다른 패키지 정보도 필요하신가요?

**Reply by cheolhui-lee** [reply]
아닙니다. 충분합니다.
번거로우시겠지만 혹시 유니티를 종료한 후,
프로젝트 디렉토리 내 Library 폴더를 삭제한 뒤 다시 유니티를 열어봐주실 수도 있을까요?

**Reply by kihye** [reply]
라이브러리 삭제 후 프로젝트 다시 열고 재시도중인데 3분정도 지났지만 아직까지 별다른 진전은 안보이네요

**Reply by cheolhui-lee** [reply]
프로젝트를 열면 아무 액션이 없는 상태에서 저 프로그레스 팝업이 계속 떠 있는 상황일까요?

**Reply by kihye** [reply]
아뇨아뇨 프로젝트를 사용하는 것 자체는 문제가 없는데 리더보드를 사용하기 위해서 Get Leaderboard Info 누르고 나서의 상황이 다를게 없다는 말이었습니다

**Reply by cheolhui-lee** [reply]
그럼 혹시 계정 로그인도 되어 있는 상태인지도 확인 해주실 수 있을까요?

**Reply by kihye** [reply]
제페토 로그인은 되어있는 상태입니다

**Reply by cheolhui-lee** [reply]
넵. 확인해 주셔서 고맙습니다.
내부에서 재현해 보고 해결될 수 있도록 하겠습니다.
번거롭게 해드려 정말 죄송합니다.

**Reply by kihye** [reply]
아닙니다 ㅎㅎ 어차피 해결되어야 할 문제니 번거로운 부분은 없습니다

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요.

리더보드 추가 로딩 문제와 관련하여 문제를 파악하여 다음 배포때 수정된 버전이 배포될 예정입니다. 
배포 이전까지 아래와 같은 방법으로 리더보드를 로딩 없이 불러오실 수 있습니다. 

### 1. Hierarchy > Create Empty 를 추가한 후 이름을 GetLeaderBoard라고 지어주세요.
<img width="282" alt="스크린샷 2022-03-04 오후 4 54 32" src="https://user-images.githubusercontent.com/100118650/156722375-189fa64b-2b1d-422c-aad7-85e0a163c6ec.png">

### 2. Project > Zepeto > TypeScript를 생성 후 이름을 GetLeaderBoard라고 지어주세요.
### 3. 아래의 내용을 복사 & 붙여넣기 해 주세요.

import { Debug } from 'UnityEngine';
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import { GetAllLeaderboardsResponse, LeaderboardAPI } from 'ZEPETO.Script.Leaderboard'

export default class GetAllLeaderboardsExample extends ZepetoScriptBehaviour {

    Start() {
        LeaderboardAPI.GetAllLeaderboards(this.OnResult, this.OnError);
    }

    OnResult(result: GetAllLeaderboardsResponse) {
        Debug.Log(`[GetAllLeaderboards] result.isSuccess: ${result.isSuccess}`);

        if (result.leaderboards) {
            for (let i = 0; i < result.leaderboards.length; ++i) {
                var leaderboard = result.leaderboards[i];
                Debug.Log(`i: ${i}, id: ${leaderboard.id}, name: ${leaderboard.name}`);
            }
        }
    }

    OnError(error: string) {
        Debug.LogError(`[GetAllLeaderboards] ${error}`);
    }
}

### 4. 스크립트를 오브젝트에 드래그해서 연결시켜 주세요.
<img width="350" alt="스크린샷 2022-03-04 오후 4 57 20" src="https://user-images.githubusercontent.com/100118650/156722743-16a21244-e7fe-4d05-b36b-02c413e11f99.png">

### 5. 상단의 **Play 버튼**을 눌러 코드가 한 번 실행되게 해 주세요. 그 후 Play는 끄셔도 됩니다.
<img width="1139" alt="스크린샷 2022-03-04 오후 4 59 18" src="https://user-images.githubusercontent.com/100118650/156723056-707a3642-aef4-40f6-8da2-dbd587e2fa5e.png">

### 6. Edit > Project Settings > Zepeto > Zepeto Leaderboard 메뉴에 들어가 주세요.
### 7. Get Leaderboard Info 버튼을 누르면 로딩 없이 리더보드가 추가됩니다 🙌

<img width="628" alt="스크린샷 2022-03-04 오후 5 02 55" src="https://user-images.githubusercontent.com/100118650/156723834-5449b089-24e7-4f6a-afb0-1c8c26a65732.png">

아래 화면이 보이면 성공! 

<img width="628" alt="스크린샷 2022-03-04 오후 5 03 04" src="https://user-images.githubusercontent.com/100118650/156723936-915181ee-0d69-4130-9294-74221548bfb1.png">

**Reply by kihye** [reply]
감사합니다. 이 방법으로 하니 문제없이 바로 됩니다.
